internal/etcd/etcd.go

```diff
@@ -0,0 +1,263 @@
+			signals <- syscall.SIGHUP
+			time.Sleep(time.Duration(1) * time.Second)
+		}
+	}
+}
+
+func NewEtcdClient(urls string, folder string) *EtcdClient {
+	// Create a new etcd client
+	cfg := client.Config{
+		Endpoints: []string{"http://127.0.0.1:2379"},
+		Transport: client.DefaultTransport,
+	}
+
+	e := &EtcdClient{}
+	c, err := client.New(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	kapi := client.NewKeysAPI(c)
+
+	e.Kapi = kapi
+	e.Folder = folder
+
+	return e
+}
+
+func (e *EtcdClient) WriteConfigDir(configdir string) error {
+	directoryEntries, err := ioutil.ReadDir(configdir)
+	if err != nil {
+		return err
+	}
+	for _, entry := range directoryEntries {
+		name := entry.Name()
+		if entry.IsDir() {
+			if name == "labels" {
+				// Handle labels
+				directoryEntries, err := ioutil.ReadDir(path.Join(configdir, name))
+				if err != nil {
+					return err
+				}
+				for _, entry := range directoryEntries {
+					filename := entry.Name()
+					if len(filename) < 6 || filename[len(filename)-5:] != ".conf" {
+						continue
+					}
+					label := filename[:len(filename)-5]
+					err = e.WriteLabelConfig(label, path.Join(configdir, name, filename))
+					if err != nil {
+						return err
+					}
+				}
+			} else if name == "hosts" {
+				// Handle hosts specific config
+				directoryEntries, err := ioutil.ReadDir(path.Join(configdir, name))
+				if err != nil {
+					return err
+				}
+
+				for _, entry := range directoryEntries {
+					filename := entry.Name()
+					if len(filename) < 6 || filename[len(filename)-5:] != ".conf" {
+						continue
+					}
+					hostname := filename[:len(filename)-5]
+					err = e.WriteHostConfig(hostname, path.Join(configdir, name, filename))
+					if err != nil {
+						return err
+					}
+				}
+			}
+			continue
+		}
+		if name == "main.conf" {
+			// Handle main config
+			err := e.WriteMainConfig(path.Join(configdir, name))
+			if err != nil {
+				return err
+			}
+		} else {
+			continue
+		}
+	}
+
+	return nil
+}
+
+func (e *EtcdClient) WriteMainConfig(path string) error {
+	// Write main config file in etcd
+	key := "main"
+	err := e.WriteConfig(key, path)
+	return err
+}
+
+func (e *EtcdClient) WriteLabelConfig(label string, path string) error {
+	// Write label config file in etcd
+	key := "labels/" + label
+	err := e.WriteConfig(key, path)
+	return err
+}
+
+func (e *EtcdClient) WriteHostConfig(host string, path string) error {
+	// Write host config file in etcd
+	key := "hosts/" + host
+	err := e.WriteConfig(key, path)
+	return err
+}
+
+func (e *EtcdClient) WriteConfig(relative_key string, path string) error {
+	// Read config file, get conf in tomlformat, convert to json
+	// Then write it to etcd
+	// TODO: Maybe we just want to store toml in etcd ? Is json really needed ????
+	// Read file
+	raw_data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	// Get toml
+	var data interface{}
+	_, err = toml.Decode(string(raw_data), &data)
```

Comment:
I believe you can use influxdata/config (https://github.com/influxdata/config/blob/master/legacy.go#L14) here